lib/core/api.js

```diff
@@ -360,6 +380,53 @@ api.upload.toPlatform = function(data, sessionInfo, progress, groupId, cb) {
   });
 };
 
+/*
+ * process the data sending it to the dataservices platform in blocks and feed back progress to the calling function
+ */
+api.upload.toDataServices = function(data, sessionInfo, progress, groupId, cb) {
```

Comment:
As part of the refactoring I've removed `api.upload.toDataServices`, passing in `uploadType` into `api.upload.toPlatform` instead to keep things DRY.